bin/metrics-per-process.py

```diff
@@ -89,6 +90,21 @@
 'CLOSING',
 'NONE'
 ]
+HANDLER_STATS = {
```

Comment:
seems like there is a much simpler way to do this, if it matches the regex just call it in the loop (no need for a lambda)  otherwise just move on. It's not that I am opposed to using lambda's when it makes sense this just seems overly complex without any real benefit over something more straightforward.

Reply:
I also wasn't sure about which way to go. Without lambdas would be simpler, but it would replace the loop with a large collection of `if`s like: 
```
if metrics_regexp.match('cpu.user'):
  stats['cpu.user'] = process_handler.cpu_times().user
```
It's not as DRY as the lambdas because the stat name is repeated, as well as the "pattern" of checking the match and storing in the dictionary. If the loss in simplicity is not worth the gain, I have no problem going for the collection of `if`s.

Reply:
I tend to write really good python in my ruby so I have never really been opposed to a bunch of if statements, much easier to follow and less rubyisms (not that lambdas are really a rubyism) are I think desired in plugins where possible. Ideally with plugins I try to write it in the most simplest of forms. This means that traditional nocs, sysadmins, etc that do not really want to learn a ton of coding can still follow and fix bugs but that might just be me. At my work I take a similar stance for a very different reason, if I need to think about it at 3 AM (when I care the most about fixing an issue quickly so I can go back to bed) when I was just pulled out of bed does it make sense, if not it either needs some healthy comments explaining what is going on or it needs to be re-written.

Reply:
+1 for comments on "tricky" looking code.

Reply:
Makes total sense, I'll go for the `if` statements.